cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/AbstractInMemoryProgramRunner.java

```diff
@@ -84,7 +87,7 @@ protected final ProgramController startAll(Program program, ProgramOptions optio
         components.put(program.getName(), instanceId, controller);
       }
 
-      return new InMemoryProgramController(components, program, options, runId);
+      return new InMemoryProgramController(components, program, runId, options, programStateWriter);
```

Comment:
I was following the idea that a program and its `runId` should precede `options` since that variable refers to a program run.